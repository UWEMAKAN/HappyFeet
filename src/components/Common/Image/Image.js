import React from 'react';
import propTypes from 'prop-types';

import './Image.css';

const Image = ({ image, title }) => (
  <img className="Image" src={image} alt={title} />
);

Image.propTypes = {
  image: propTypes.string.isRequired,
  title: propTypes.string.isRequired
};

export default Image;
