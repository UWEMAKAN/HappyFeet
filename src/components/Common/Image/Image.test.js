import React from 'react';
import { shallow } from 'enzyme';
import Image from './Image';

const renderImage = (args) => {
  const defaultProps = {
    image: '',
    title: ''
  };
  const props = { ...defaultProps, ...args };
  // eslint-disable-next-line react/jsx-props-no-spreading
  return shallow(<Image {...props} />);
};

describe('Image Component', () => {
  test('should render image component', () => {
    const image = 'http://localhost:3001/assets/images/image.jpg';
    const title = 'Nike Black Snickers for men and women';
    const wrapper = renderImage({ image, title });

    expect.assertions(4);
    expect(wrapper.find('img').length).toBe(1);
    expect(wrapper.prop('src')).toEqual(image);
    expect(wrapper.prop('alt')).toEqual(title);
    expect(wrapper).toMatchSnapshot();
  });
});
