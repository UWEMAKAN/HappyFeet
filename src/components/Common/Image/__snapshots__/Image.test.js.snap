// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component should render image component 1`] = `ShallowWrapper {}`;
